.drone.yml

```diff
@@ -1,297 +1,391 @@
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     environment:
-      - ENVIRONMENT=wcs-dev
-      - VERSION=build-${DRONE_BUILD_NUMBER}
-      - HOCS_INFO_SERVICE_DATA_VERSION=latest
+      - ENVIRONMENT=wcs-qa
       - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
     secrets:
-      - hocs_info_service_dev_wcs
+      - hocs_info_service_cs_qa
     commands:
+      - source version.txt
+      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    depends_on:
+      - clone kube repo
+      - WCS downstream build
     when:
-      branch: master
-      event: [push, tag]
+      event:
+        - promote
+      target:
+        - release
+
 
-  deployment:
+  - name: deploy to cs-dev
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
-    environment:
-      - ENVIRONMENT=${DRONE_DEPLOY_TO}
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_info_service_dev_cs
-      - hocs_info_service_demo_cs
-      - hocs_info_service_dev_wcs
-      - hocs_info_service_demo_wcs
     commands:
       - cd kube-hocs-info-service
       - ./deploy.sh
+    environment:
+      ENVIRONMENT: cs-dev
+      KUBE_TOKEN:
+        from_secret: hocs_info_service_cs_dev
+      KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
+      VERSION: build_${DRONE_BUILD_NUMBER}
     when:
-      event: deployment
-      environment: [cs-dev, cs-demo, wcs-dev, wcs-demo]
+      branch:
+        - main
+      event:
+        - push
+    depends_on:
+      - clone kube repo
 
-  deploy-to-qa-cs:
+  - name: deploy to wcs-dev
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
-    environment:
-      - ENVIRONMENT=cs-qa
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_info_service_qa_cs
     commands:
-      - source version.txt
-      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    environment:
+      ENVIRONMENT: wcs-dev
+      KUBE_TOKEN:
+        from_secret: hocs_info-service_wcs_dev
+      VERSION: build_${DRONE_BUILD_NUMBER}
     when:
-      event: deployment
-      environment: cs-qa
+      branch:
+        - main
+      event:
+        - push
+    depends_on:
+      - clone kube repo
 
-  deploy-to-qa-wcs:
+  - name: deployment
     image: quay.io/ukhomeofficedigital/kd:v1.16.0
     environment:
-      - ENVIRONMENT=wcs-qa
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_info_service_qa_wcs
+      ENVIRONMENT: ${DRONE_DEPLOY_TO}
+      KUBE_TOKEN:
+        from_secret: hocs_info_service_${DRONE_DEPLOY_TO/-/_}
     commands:
-      - source version.txt
-      - echo $VERSION
       - cd kube-hocs-info-service
       - ./deploy.sh
+    depends_on:
+      - clone kube repo
     when:
-      event: deployment
-      environment: wcs-qa
+      target:
+        exclude:
+          - release
+          - restart
```

Comment:
Probably worth excluding this from other steps too - as it doesn't impact this pipeline at all.

Reply:
See how this goes - doesn't need doing yet.